base/expr.jl

```diff
@@ -9,7 +9,8 @@ const is_expr = isexpr
 """
     gensym([tag])
 
-Generates a symbol which will not conflict with other variable names (in the same module).
+Generate a symbol unique among all calls to this function within the same process.
```

Comment:
Note that it is definitely not unique among names in the process (as the failed tests show), but only from context of defining one module

Reply:
That's why I said "calls within the process". What I'm trying to say is that all calls to this within one process will return different names (different from each other, not necessarily different from anything else). I think that's all we can promise? I don't see how it involves modules except in that packages are precompiled in separate processes.

Reply:
I think that's a lot more accurate than connecting it with modules, since you can run into this problem *within* one module, not just between them.